projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item-list/cart-item-list.component.ts

```diff
@@ -1,57 +1,73 @@
-import { Component, Input, OnInit } from '@angular/core';
-import { FormBuilder, FormGroup } from '@angular/forms';
+import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
+import { FormControl, FormGroup } from '@angular/forms';
 import { CartService, PromotionResult } from '@spartacus/core';
+import { Observable } from 'rxjs';
+import { map, startWith } from 'rxjs/operators';
 import { Item } from '../cart-item/cart-item.component';
 
 @Component({
   selector: 'cx-cart-item-list',
   templateUrl: './cart-item-list.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class CartItemListComponent implements OnInit {
-  @Input()
-  isReadOnly = false;
+export class CartItemListComponent {
+  @Input() isReadOnly = false;
 
-  @Input()
-  hasHeader = true;
+  @Input() hasHeader = true;
 
-  @Input()
-  items: Item[] = [];
+  @Input('items')
+  // TODO: currently we're getting a new array of items if the cart changes. pretty annoying as
+  // it forces a repaint on the screen, which is noticable in the UI.
+  set items(items: Item[]) {
+    this._items = items;
+    this.createForm(items);
+  }
+  get items(): Item[] {
+    return this._items;
+  }
 
-  @Input()
-  potentialProductPromotions: PromotionResult[] = [];
+  @Input() potentialProductPromotions: PromotionResult[] = [];
 
-  @Input()
-  cartIsLoading = false;
+  _loading: boolean;
```

Comment:
```suggestion
  private _loading: boolean;
```

Is it needed at all?

Reply:
nope, thx